lib/slim-rails.rb

```diff
@@ -8,10 +8,14 @@ class Railtie < ::Rails::Railtie
       config.app_generators.template_engine :slim
 
       initializer 'slim_rails.configure_template_digestor' do |app|
-        if config.respond_to?(:assets)
-          config.assets.configure do |env|
-            env.assets.register_engine '.slim', Slim::Template
+        if ::Rails::VERSION::MAJOR == 3
+          if app.assets && app.assets.respond_to?(:register_engine)
+            app.assets.register_engine('.slim', Slim::Template)
           end
+        else
+          config.assets.configure do |env|
+            env.register_engine('.slim', Slim::Template)
+          end if config.respond_to?(:assets)
         end
```

Comment:
> What issue are you trying to resolve here?

In Rails 4, `rake assets:precompile` doesn't output html assets for .slim.html in slim-rails 3.1.0.

I create a reproduce code for that.

Add a app/views/html/404.html.slim, then `rake assets:precompile`.

Rails 3.2:
https://github.com/hanachin/slim-assets-precompile-sample/blob/master/Dummy3/public/assets/404-2e16c2d883028a2478614e4642046d4f.html

Rails 4.2:
https://github.com/hanachin/slim-assets-precompile-sample/blob/master/Dummy42/public/assets/404.html-20f82e938662011919dd34ce33f5332a8197b69fcb99109f0425d005758429e7.slim